kong/db/schema/entities/routes.lua

```diff
@@ -2,18 +2,35 @@ local typedefs = require("kong.db.schema.typedefs")
 local router = require("resty.router.router")
 local deprecation = require("kong.deprecation")
 
-local CACHED_SCHEMA = require("kong.router.atc").schema
-local get_expression = require("kong.router.compat").get_expression
+local validate_entity_by_expression
+local has_paths
+do
+  local isempty        = require("table.isempty")
+  local CACHED_SCHEMA  = require("kong.router.atc").schema
+  local get_expression = require("kong.router.compat").get_expression
 
-local function validate_expression(id, exp)
-  local r = router.new(CACHED_SCHEMA)
+  local type = type
 
-  local res, err = r:add_matcher(0, id, exp)
-  if not res then
-    return nil, "Router Expression failed validation: " .. err
+  local r = router.new(CACHED_SCHEMA, 1)
+
+  validate_entity_by_expression = function(entity)
+    local id = entity.id
+    local exp = entity.expression or get_expression(entity)
+
+    local res, err = r:add_matcher(0, id, exp)
```

Comment:
Could we add a new API like `r:validate()` in atc_router? 

In my opinion, `r:add_matcher` is not the best idea to validate expression.

Reply:
We should implement this API in atc-router's Lua code.

Reply:
need to refactor after https://github.com/Kong/atc-router/pull/39 merged.